go/nbs/store.go

```diff
@@ -290,13 +304,24 @@ func (nbs *NomsBlockStore) UpdateRoot(current, last hash.Hash) bool {
 		nbs.mt = nil
 	}
 
-	actual, tableNames := nbs.mm.Update(nbs.tables.ToSpecs(), nbs.root, current, nil)
+	candidate := nbs.tables
+	var compactees chunkSources
+	if candidate.Size() > nbs.maxTables {
```

Comment:
No, `Size()` is just returning the number of tables, which is unaffected by compacting a memTable (do we need a different term for this now?). A `tableSet` is immutable; nothing changes the number of tables in a set after it's constructed.